<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>M_W</class>
 <widget class="QMainWindow" name="M_W">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>834</width>
    <height>600</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>N Queens Problem (GA)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>5</number>
       </property>
       <property name="topMargin">
        <number>5</number>
       </property>
       <property name="rightMargin">
        <number>5</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item row="0" column="0">
        <widget class="QFrame" name="frame_2">
         <property name="minimumSize">
          <size>
           <width>500</width>
           <height>200</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>400</width>
           <height>200</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <widget class="QLabel" name="label">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>20</y>
            <width>131</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string> Queens Number :</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="qnum">
          <property name="geometry">
           <rect>
            <x>170</x>
            <y>20</y>
            <width>41</width>
            <height>31</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>13</pointsize>
           </font>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="maxLength">
           <number>3</number>
          </property>
          <property name="frame">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QPushButton" name="run_btn">
          <property name="geometry">
           <rect>
            <x>370</x>
            <y>60</y>
            <width>91</width>
            <height>31</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <family>Segoe UI Semibold</family>
            <pointsize>15</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(0, 255, 255);</string>
          </property>
          <property name="text">
           <string>RUN</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>60</y>
            <width>121</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string> Population Size :</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="pop_len">
          <property name="geometry">
           <rect>
            <x>170</x>
            <y>60</y>
            <width>71</width>
            <height>31</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>13</pointsize>
           </font>
          </property>
          <property name="text">
           <string>200</string>
          </property>
          <property name="maxLength">
           <number>20</number>
          </property>
          <property name="frame">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_3">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>100</y>
            <width>121</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string> Mutation Rate :</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="mut_rate">
          <property name="geometry">
           <rect>
            <x>170</x>
            <y>100</y>
            <width>31</width>
            <height>31</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>13</pointsize>
           </font>
          </property>
          <property name="text">
           <string>2</string>
          </property>
          <property name="maxLength">
           <number>2</number>
          </property>
          <property name="frame">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_4">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>140</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string> Max Iteration :</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="max_iter">
          <property name="geometry">
           <rect>
            <x>170</x>
            <y>140</y>
            <width>71</width>
            <height>31</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>13</pointsize>
           </font>
          </property>
          <property name="text">
           <string>10000</string>
          </property>
          <property name="maxLength">
           <number>20</number>
          </property>
          <property name="frame">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_5">
          <property name="geometry">
           <rect>
            <x>200</x>
            <y>100</y>
            <width>21</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string> %</string>
          </property>
         </widget>
         <widget class="QPushButton" name="stop_btn">
          <property name="geometry">
           <rect>
            <x>370</x>
            <y>110</y>
            <width>91</width>
            <height>31</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <family>Segoe UI Semibold</family>
            <pointsize>15</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color: rgb(255, 0, 0);</string>
          </property>
          <property name="text">
           <string>STOP</string>
          </property>
         </widget>
        </widget>
       </item>
       <item row="1" column="0" rowspan="2">
        <widget class="QPlainTextEdit" name="log_output">
         <property name="minimumSize">
          <size>
           <width>500</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>500</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="2" rowspan="2">
        <widget class="QLabel" name="log_shape">
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>300</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>700</width>
           <height>700</height>
          </size>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPlainTextEdit" name="log_queens">
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>100</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>700</width>
           <height>100</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
